python_transport/wirepas_gateway/protocol/mqtt_wrapper.py

```diff
@@ -149,24 +163,27 @@ def _get_socket(self):
             self.logger.error("Impossible to connect - authentication failure ?")
             return None
 
-        # Socket is not opened anymore, try to reconnect
-        timeout = MQTTWrapper.TIMEOUT_RECONNECT_S
-        while timeout > 0:
+        # Socket is not opened anymore, try to reconnect for timeout if set
+        loop_forever = self.timeout == 0
+        delay = 0
+
+        # Loop forever or until timeout is over
+        while loop_forever or (delay <= self.timeout):
             try:
                 ret = self._client.reconnect()
                 if ret == mqtt.MQTT_ERR_SUCCESS:
```

Comment:
I think we could improve this on a separate PR, for example, I think there is no point to retry if the error signals an auth issue.

Reply:
It should already be handled as it will not loop if there were no successful connection yet (probably the case if auth isue).
But agree that it can be improved